File: src/main/kotlin/com/io/reactivecoroutines/weather/WeatherCityController.java
package com.io.reactivecoroutines.weather;

import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;

@RestController
@Validated
@RequestMapping("/weather-infos/cities")
public class WeatherCityController {
    private final WeatherService weatherService;

    public WeatherCityController(WeatherService weatherService) {
        this.weatherService = weatherService;
    }

    @GetMapping
    public Flux<WeatherInfo> getAllWeatherByCity(@RequestParam String city) {
        return weatherService.queryWeatherInfoByCity(city);
    }
}

File: src/main/kotlin/com/io/reactivecoroutines/weather/WeatherCityController.kt
package com.io.reactivecoroutines.weather

import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@Validated
@RequestMapping("/weather-infos/cities/")
class WeatherCityController(
    private val weatherService: WeatherService
) {

    @GetMapping
    suspend fun getAllWeatherByCity(@RequestParam city: String): List<WeatherInfo> {
        return weatherService.queryWeatherInfoByCity(city)
    }
}
